gnosis/cowswap
# [ABA 🎵 ] - Base PR: labels, logic, optimisations

# Summary

Part of #1495 - waterfalls into the base #1872 

### THIS PR FORCES ABA IMPACT CALC EVERYTIME! 
_it ignores fiat price impact. in the final PR it should first consider fiat impact, then aba. in another another pr we will consider a message when the latter and former fail._

Includes a simple test.

## QA:
- feel free to just test this as is, good to see what comes back also on xdai. if nothing, it's because the inverse B > A trade doesnt exist. This can be verifiied in the Networks tab as a red query and the `NoLiquidity` error as a response
- btw - the merge of #1896 will make this experience easier as well - rn no liquidty just returns a generic try again error

  # To Test

1. head to mainnet
2. select any tokens
3. input amounts
4. see impact

## NOTES
- i noticed that on BUY orders ABA simulations we often get 0% - i've checked both the AB_Trade and BA_Trade objects and they are correct - as in the AB_Trade.tradeType returns 1 (BUY) and the BA_Trade.tradeType returns 0 (SELL) - and the inputs/outputs are the same for both. So ABA price impact for buy orders seems much less accurate (or maybe more?)
